pkg/questionModel/keyValue.go

```diff
@@ -0,0 +1,247 @@
+		kv.tags = append(kv.tags, []string{strings.TrimSpace(pair[0]), strings.TrimSpace(pair[1])})
+	}
+
+	tagList := &SingleSelectList{}
+	tagList.InitializeModel(&BubbleTeaData{
+		OptionData:    kv.tags,
+		HeaderStrings: []string{"KEY", "VALUE"},
+	})
+	tagList.list.Select(-1)
+
+	kv.tagList = tagList
+	kv.question = data.QuestionString
+}
+
+func (kv *keyValue) Init() tea.Cmd {
+	return textinput.Blink
+}
+
+func (kv *keyValue) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	switch msg := msg.(type) {
+	case tea.KeyMsg:
+
+		switch msg.Type {
+		case tea.KeyCtrlC:
+			kv.err = errors.New("User has quit before finishing question!")
+			return kv, tea.Quit
+
+		case tea.KeyUp, tea.KeyDown, tea.KeyEnter, tea.KeyShiftTab, tea.KeyTab:
+			msgType := msg.Type
+
+			if msgType == tea.KeyEnter && kv.focusIndex == len(kv.inputs) {
+				if !kv.submitButtonFocused {
+					kv.addTag()
+				} else {
+					return kv, tea.Quit
+				}
+			}
+
+			kv.cycleFocusIndex(msgType)
+
+			if kv.focusIndex != len(kv.inputs) {
+				kv.submitButtonFocused = false
+			}
+
+			cmds := make([]tea.Cmd, len(kv.inputs))
+			for i := 0; i <= len(kv.inputs)-1; i++ {
+				if i == kv.focusIndex {
+					cmds[i] = kv.focusInput(i)
+					continue
+				}
+				kv.inputs[i].Blur()
+				kv.inputs[i].PromptStyle = smallLeftPadding
+				kv.inputs[i].TextStyle = noStyle
+			}
+
+			return kv, tea.Batch(cmds...)
+
+		case tea.KeyRight, tea.KeyLeft:
+			if kv.focusIndex == len(kv.inputs) {
+				kv.submitButtonFocused = !kv.submitButtonFocused
+			}
+
+		case tea.KeyBackspace:
+			kv.deleteTag()
+			if len(kv.tagList.list.Items()) == 0 {
+				kv.focusIndex = 0
+				return kv, kv.focusInput(kv.focusIndex)
+			}
+		}
+
+	case error:
+		kv.err = msg
+		return kv, tea.Quit
+	}
+
+	cmd := kv.updateInputs(msg)
+
+	return kv, cmd
+}
+
+func (kv *keyValue) View() string {
+	var b strings.Builder
+	if kv.question != "" {
+		b.WriteString(kv.question + "\n")
+	}
+
+	if len(kv.tagList.list.Items()) > 0 {
+		b.WriteString("\n" + kv.tagList.View() + "\n")
+	} else {
+		b.WriteRune('\n')
+	}
+
+	for i := range kv.inputs {
+		b.WriteString(kv.inputs[i].View())
+		if i < len(kv.inputs)-1 {
+			b.WriteRune('\n')
+		}
+	}
+
+	addButton := kv.createButton(addButtonText, !kv.submitButtonFocused)
+	submitButton := kv.createButton(submitButtonText, kv.submitButtonFocused)
+	b.WriteString(fmt.Sprintf(smallLeftPadding.Render("\n\n%s  %s\n"), addButton, submitButton))
+
+	return b.String()
+}
+
+func (kv *keyValue) addTag() {
+	if kv.inputs[0].Value() == "" {
+		kv.inputs[0].Placeholder = "Please Enter A Key!"
+	}
+
+	if kv.inputs[1].Value() == "" {
+		kv.inputs[1].Placeholder = "Please Enter A Value!"
+	}
+
+	if kv.inputs[0].Value() != "" && kv.inputs[1].Value() != "" {
+		kv.tags = append(kv.tags, []string{strings.TrimSpace(kv.inputs[0].Value()), strings.TrimSpace(kv.inputs[1].Value())})
+		kv.tagList.InitializeModel(&BubbleTeaData{
+			OptionData:    kv.tags,
+			HeaderStrings: []string{"KEY", "VALUE"},
+		})
```

Comment:
Is it necessary to initialize the model again? If so, can we extract the value of `HeaderStrings` into a constant to avoid shotgun surgery with line 57 if the header names ever change?